### DISCUSSION TITLE
Decoder -> Encoder?

### DISCUSSION THREAD

**Discussion by jamesosaurus** [image-rs/image-png#344] [Q&A] [answered]
Is there a standard way of writing out a png that has previously been read in (presumably after some alteration)?

**Comment by fintelia** [comment, marked as answer]
There isn't one specific way. But something like this should be enough:

```rust
let img = io::Reader::open("original.png")?.decode()?;
// ... modify img ...
img.save("modified.png")?;
```

**Reply by jamesosaurus** [reply]
@fintelia Thanks for the reply.  I am still a bit confused in that this seems like a different crate (image vs png).  Unfortunately, I was specifically interested in the API in png due to metadata support.  Sorry if I am missing the obvious-  I am just learning Rust.

I am assuming that this discussion area is associated to https://docs.rs/png/latest/png/index.html as this is the repository linked to from there.

**Reply by fintelia** [reply]
Ah, sorry I missed that. The image crate just calls into the png crate, so if you're looking for a simple approach I'd recommend that. It is a bit more involved to use the png crate directly
